lib/V8/v8-utils.cpp

```diff
@@ -3044,6 +3044,45 @@ static void JS_Read(v8::FunctionCallbackInfo<v8::Value> const& args) {
   TRI_V8_TRY_CATCH_END
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief was docuBlock JS_Read
+////////////////////////////////////////////////////////////////////////////////
+
+static void JS_ReadPipe(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate)
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("read(pipefd)");
+  }
+
+  int64_t f = TRI_ObjectToInt64(isolate, args[0]);
+  if (f == 0) {
+    TRI_V8_THROW_TYPE_ERROR("<filename> a file pointer");
+  }
+
+  // TRI_GET_GLOBALS();
+  // V8SecurityFeature& v8security = v8g->_server.getFeature<V8SecurityFeature>();
+  // 
+  // if (!v8security.isAllowedToAccessPath(isolate, *name, FSAccessType::READ)) {
+  //   THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_FORBIDDEN,
+  //                                  std::string("not allowed to read files in this path: ") + *name);
+  // }
```

Comment:
is this needed? I think we need to enable the code.

Reply:
it should probably check this against the list of spawned processes filedescriptors. 
I don't know yet how this will work on windows, since that is using something different - whether and how I'll be able to translate this back and forth from the js data

Reply:
changed the code to use the PID. If the PID is found in the list of spawned processes, the user has obviously been able to spawn this process, and then may read from its PID.  Hence adding the process spawn security profile.